File: Analysis/meftools/R/mef_info.R
mef_info <- function( args ) {
  #' Return a structure containing MEF header information.
  #' Based on Matlab code "read_mef_discont".
  #' 
  #' @param filename String: The complete path to a .mef file
  #' @param password String: The public password for the MEF file.
  #' @export
  library( meftools )
  
  options( warn = -1 )
  
  filename <- args[1]
  password <- args[2]

  # Read the headerpwd
  print( paste0( filename, ' ', password ) )
  header = read_mef_header(c(filename,password))
#  print( header )
  
  # Read the ToC
  # Use Rcpp to pass in 'index_data_offset' and 'number_of_index_entries', then
  # return the ToC.
  print( 'Reading ToC' )
  variables <- c(filename,header$index_data_offset,header$number_of_index_entries)
  print( paste0( variables ) )
  ToC <- table_of_contents( variables )
  print( 'Read ToC' )

  # Read the discontnuities
  # Use Rcpp to pass in 'number_of_index_entries', then
  # return the discontinuity flags.
  print( 'Reading discontinuities' )
  discontinuities <- get_discontinuities( c(filename,header$number_of_index_entries), ToC );
  print( 'Read discontinuities' )
  
  # Put the returnable values in a named list.
  value <- list( header=header, ToC=ToC, discontinuities=discontinuities );  
  return( value )
}

File: Analysis/meftools/C/libmeftools/old/mef_info.cpp






std::map mef_info( char* filename, char* password ) {
	Rcpp::StringVector strings(2);
// Pack the strings
	strings(0) = filename;
	strings(1) = password;

// Prepare memory space
	MEFTOOLS::MEF_HEADER_INFO mef_info;

// Read the header
	Rcpp::MEF_HEADER_INFO rcpp_mef_header_info = read_mef_header( strings );

// Unpack the structure
	MEF_HEADER mef_header = convertRcppHeader( rcpp_mef_header_info );

// Return
	return( mef_info );
}


